File: services/education/internal/model/class_record.go
package model

import "time"

type StudentTeachingRecordPagedQueryDTO struct {
	PageRequestModel RollCallPageRequestModel        `json:"pageRequestModel"`
	SortModel        StudentTeachingRecordSortModel  `json:"sortModel"`
	QueryModel       StudentTeachingRecordQueryModel `json:"queryModel"`
}

type TeachingRecordDetailQueryDTO struct {
	TeachingRecordID string `json:"teachingRecordId"`
}

type UpdateTeachingRecordClassInfoDTO struct {
	TeachingRecordID string   `json:"teachingRecordId"`
	TeacherID        string   `json:"teacherId"`
	AssistantIDs     []string `json:"assistantIds"`
	ClassRoomID      string   `json:"classRoomId"`
	TeacherClassTime float64  `json:"teacherClassTime"`
}

type UpdateStudentTeachingRecordDTO struct {
	StudentTeachingRecordID string  `json:"studentTeachingRecordId"`
	TeachingRecordID        string  `json:"teachingRecordId"`
	StudentID               string  `json:"studentId"`
	SourceType              int     `json:"sourceType"`
	Status                  int     `json:"status"`
	Quantity                float64 `json:"quantity"`
	Remark                  string  `json:"remark"`
	ExternalRemark          string  `json:"externalRemark"`
}

type DeleteStudentTeachingRecordDTO struct {
	StudentTeachingRecordID string `json:"studentTeachingRecordId"`
}

type DeleteTeachingRecordDTO struct {
	TeachingRecordID string `json:"teachingRecordId"`
}

type StudentTeachingRecordQueryModel struct {
	BeginStartTime                string   `json:"beginStartTime"`
	EndStartTime                  string   `json:"endStartTime"`
	BeginCreateTime               string   `json:"beginCreateTime"`
	EndCreateTime                 string   `json:"endCreateTime"`
	BeginUpdatedTime              string   `json:"beginUpdatedTime"`
	EndUpdatedTime                string   `json:"endUpdatedTime"`
	StudentID                     string   `json:"studentId"`
	TeacherIDs                    []string `json:"teacherIds"`
	AssistantTeacherIDs           []string `json:"assistantTeacherIds"`
	One2OneIDs                    []string `json:"one2OneIds"`
	TimetableSourceTypes          []int    `json:"timetableSourceTypes"`
	StudentSourceTypes            []int    `json:"studentSourceTypes"`
	LessonChargingModeEnums       []int    `json:"lessonChargingModeEnums"`
	StudentTeachingRecordStatuses []int    `json:"studentTeachingRecordStatuses"`
	ScheduleCallStatus            *int     `json:"scheduleCallStatus,omitempty"`
	IsArrear                      *bool    `json:"isArrear"`
	LessonIDs                     []string `json:"lessonIds"`
	ClassIDs                      []string `json:"classIds"`
	StudentTeachingRecordIDs      []string `json:"studentTeachingRecordIds"`
	TeachingRecordIDs             []string `json:"teachingRecordIds"`
}

type StudentTeachingRecordSortModel struct {
	StartTime   int `json:"startTime"`
	UpdatedTime int `json:"updatedTime"`
}

type StudentTeachingRecordItem struct {
	StudentTeachingRecordID   string   `json:"studentTeachingRecordId"`
	TeachingRecordID          string   `json:"teachingRecordId"`
	StudentID                 string   `json:"studentId"`
	StudentName               string   `json:"studentName"`
	StudentPhone              string   `json:"studentPhone"`
	Avatar                    string   `json:"avatar"`
	TeacherName               string   `json:"teacherName"`
	TeacherEmployeeType       int      `json:"teacherEmployeeType"`
	Assistants                string   `json:"assistants"`
	ClassName                 string   `json:"className"`
	One2OneName               string   `json:"one2OneName"`
	LessonName                string   `json:"lessonName"`
	Status                    int      `json:"status"`
	SourceType                int      `json:"sourceType"`
	StartTime                 string   `json:"startTime"`
	EndTime                   string   `json:"endTime"`
	TeachingRecordCreatedTime string   `json:"teachingRecordCreatedTime"`
	TimetableSourceType       int      `json:"timetableSourceType"`
	UpdatedTime               string   `json:"updatedTime"`
	UpdatedStaffName          string   `json:"updatedStaffName"`
	RecordTime                string   `json:"recordTime"`
	Quantity                  float64  `json:"quantity"`
	ActualQuantity            float64  `json:"actualQuantity"`
	Amount                    float64  `json:"amount"`
	SkuMode                   int      `json:"skuMode"`
	ActualDeduct              float64  `json:"actualDeduct"`
	ActualTuition             float64  `json:"actualTuition"`
	ArrearQuantity            float64  `json:"arrearQuantity"`
	Remark                    string   `json:"remark"`
	ExternalRemark            string   `json:"externalRemark"`
	TuitionAccountID          string   `json:"tuitionAccountId"`
	TuitionAccountName        string   `json:"tuitionAccountName"`
	HasCompensated            bool     `json:"hasCompensated"`
	SubjectID                 string   `json:"subjectId"`
	SubjectName               string   `json:"subjectName"`
	AdvisorStaffID            string   `json:"advisorStaffId"`
	AdvisorStaffName          string   `json:"advisorStaffName"`
	StudentManagerID          string   `json:"studentManagerId"`
	StudentManagerName        string   `json:"studentManagerName"`
	TeachingContent           string   `json:"teachingContent"`
	TeachingContentImages     []string `json:"teachingContentImages"`
	ClassRoomName             string   `json:"classRoomName"`
	One2OneTeachers           string   `json:"one2OneTeachers"`
	ClassTeachers             string   `json:"classTeachers"`
	RollCallClassTeachers     string   `json:"rollCallClassTeachers"`
	CurrentClassTeachers      string   `json:"currentClassTeachers"`
}

type StudentTeachingRecordPagedResult struct {
	TotalClassTimes   float64                     `json:"totalClassTimes"`
	TotalTuition      float64                     `json:"totalTuition"`
	TotalStudentCount int                         `json:"totalStudentCount"`
	List              []StudentTeachingRecordItem `json:"list"`
	Total             int                         `json:"total"`
}

type TeachingRecordDetailTeacher struct {
	TeacherID   string  `json:"teacherId"`
	TeacherName string  `json:"teacherName"`
	Type        int     `json:"type"`
	Status      int     `json:"status"`
	Quantity    float64 `json:"quantity"`
}

type TeachingRecordDetailStudent struct {
	StudentTeachingRecordID string  `json:"studentTeachingRecordId"`
	StudentID               string  `json:"studentId"`
	StudentName             string  `json:"studentName"`
	StudentPhone            string  `json:"studentPhone"`
	Avatar                  string  `json:"avatar"`
	Status                  int     `json:"status"`
	SourceType              int     `json:"sourceType"`
	Quantity                float64 `json:"quantity"`
	ActualQuantity          float64 `json:"actualQuantity"`
	Remark                  string  `json:"remark"`
	ExternalRemark          string  `json:"externalRemark"`
	TuitionAccountID        string  `json:"tuitionAccountId"`
	TuitionAccountName      string  `json:"tuitionAccountName"`
	IsTuitionAccountActive  bool    `json:"isTuitionAccountActive"`
	LeftQuantity            float64 `json:"leftQuantity"`
	SkuMode                 int     `json:"skuMode"`
	Amount                  float64 `json:"amount"`
	ActualDeduct            float64 `json:"actualDeduct"`
	ActualTuition           float64 `json:"actualTuition"`
	ArrearQuantity          float64 `json:"arrearQuantity"`
	RecordTime              string  `json:"recordTime"`
	UpdatedTime             string  `json:"updatedTime"`
	UpdatedStaffName        string  `json:"updatedStaffName"`
}

type TeachingRecordDetailResult struct {
	TeachingRecordID        string                        `json:"teachingRecordId"`
	SourceName              string                        `json:"sourceName"`
	SourceType              int                           `json:"sourceType"`
	SourceID                string                        `json:"sourceId"`
	LessonID                string                        `json:"lessonId"`
	LessonType              int                           `json:"lessonType"`
	StartTime               string                        `json:"startTime"`
	EndTime                 string                        `json:"endTime"`
	ShouldAttendanceCount   int                           `json:"shouldAttendanceCount"`
	ActualAttendanceCount   int                           `json:"actualAttendanceCount"`
	LeaveCount              int                           `json:"leaveCount"`
	TruancyCount            int                           `json:"truancyCount"`
	TeacherClassTime        float64                       `json:"teacherClassTime"`
	StudentTotalClassTime   float64                       `json:"studentTotalClassTime"`
	StudentActualTuition    float64                       `json:"studentActualTuition"`
	TeacherList             []TeachingRecordDetailTeacher `json:"teacherList"`
	StudentList             []TeachingRecordDetailStudent `json:"studentList"`
	CreatedTime             string                        `json:"createdTime"`
	CreatedStaffName        string                        `json:"createdStaffName"`
	TimetableSourceType     int                           `json:"timetableSourceType"`
	ClassRoomName           string                        `json:"classRoomName"`
	ClassRoomID             string                        `json:"classRoomId"`
	DefaultStudentClassTime float64                       `json:"defaultStudentClassTime"`
	TimetableSourceID       string                        `json:"timetableSourceId"`
	LessonName              string                        `json:"lessonName"`
	TeachingContent         string                        `json:"teachingContent"`
	SubjectID               string                        `json:"subjectId"`
	SubjectName             string                        `json:"subjectName"`
	TeachingContentImages   []string                      `json:"teachingContentImages"`
}

type ScheduleTeachingRecordPagedQueryDTO struct {
	PageRequestModel RollCallPageRequestModel        `json:"pageRequestModel"`
	SortModel        ScheduleTeachingRecordSortModel `json:"sortModel"`
	QueryModel       StudentTeachingRecordQueryModel `json:"queryModel"`
}

type ScheduleTeachingRecordSortModel struct {
	StartTime   int `json:"startTime"`
	UpdatedTime int `json:"updatedTime"`
}

type ScheduleTeachingRecordItem struct {
	TeachingRecordID    string  `json:"teachingRecordId"`
	TimetableSourceID   string  `json:"timetableSourceId,omitempty"`
	StartTime           string  `json:"startTime"`
	EndTime             string  `json:"endTime"`
	TimetableSourceType int     `json:"timetableSourceType"`
	SourceName          string  `json:"sourceName"`
	SourceType          int     `json:"sourceType"`
	SourceID            string  `json:"sourceId"`
	LessonID            string  `json:"lessonId"`
	ClassName           string  `json:"className"`
	One2OneName         string  `json:"one2OneName"`
	LessonName          string  `json:"lessonName"`
	SubjectID           string  `json:"subjectId"`
	SubjectName         string  `json:"subjectName"`
	RollCallStatus      int     `json:"rollCallStatus"`
	AttendanceRate      float64 `json:"attendanceRate"`
	AttendCount         int     `json:"attendCount"`
	ShouldAttendCount   int     `json:"shouldAttendCount"`
	LeaveCount          int     `json:"leaveCount"`
	AbsentCount         int     `json:"absentCount"`
	UnrecordedCount     int     `json:"unrecordedCount"`
	ActualQuantity      float64 `json:"actualQuantity"`
	ActualTuition       float64 `json:"actualTuition"`
	TeacherID           string  `json:"teacherId"`
	TeacherName         string  `json:"teacherName"`
	Assistants          string  `json:"assistants"`
	ClassRoomName       string  `json:"classRoomName"`
	TeacherClassTime    float64 `json:"teacherClassTime"`
	CommentCount        int     `json:"commentCount"`
	UnCommentCount      int     `json:"unCommentCount"`
	ReadCount           int     `json:"readCount"`
	UnReadCount         int     `json:"unReadCount"`
	CreatedTime         string  `json:"createdTime"`
	UpdatedTime         string  `json:"updatedTime"`
}

type ScheduleTeachingRecordPagedResult struct {
	TotalClassTimes   float64                      `json:"totalClassTimes"`
	TotalTeacherTimes float64                      `json:"totalTeacherTimes"`
	TotalTuition      float64                      `json:"totalTuition"`
	List              []ScheduleTeachingRecordItem `json:"list"`
	Total             int                          `json:"total"`
}

type ClassCommentPagedQueryDTO struct {
	PageRequestModel RollCallPageRequestModel `json:"pageRequestModel"`
	SortModel        ClassCommentSortModel    `json:"sortModel"`
	QueryModel       ClassCommentQueryModel   `json:"queryModel"`
}

type ClassCommentSortModel struct {
	StartTime int `json:"startTime"`
}

type ClassCommentQueryModel struct {
	TeachingStartTime   string   `json:"teachingStartTime"`
	TeachingEndTime     string   `json:"teachingEndTime"`
	TeachingRecordTypes []int    `json:"teachingRecordTypes"`
	LessonID            string   `json:"lessonId"`
	TeacherIDs          []string `json:"teacherIds"`
	ClassID             string   `json:"classId"`
	One2OneID           string   `json:"one2OneId"`
	ClassTeacherIDs     []string `json:"classTeacherIds"`
	One2OneTeacherIDs   []string `json:"one2OneTeacherIds"`
}

type ClassCommentPagedItem struct {
	TeachingRecordID string `json:"teachingRecordId"`
	SourceName       string `json:"sourceName"`
	SourceType       int    `json:"sourceType"`
	SourceID         string `json:"sourceId"`
	LessonID         string `json:"lessonId"`
	LessonName       string `json:"lessonName"`
	CreatedTime      string `json:"createdTime"`
	TeacherID        string `json:"teacherId"`
	TeacherName      string `json:"teacherName"`
	StartTime        string `json:"startTime"`
	EndTime          string `json:"endTime"`
	ReadCount        int    `json:"readCount"`
	UnReadCount      int    `json:"unReadCount"`
	CommentCount     int    `json:"commentCount"`
	UnCommentCount   int    `json:"unCommentCount"`
	Assistants       string `json:"assistants"`
	ClassRoomName    string `json:"classRoomName"`
}

type ClassCommentPagedResult struct {
	List  []ClassCommentPagedItem `json:"list"`
	Total int                     `json:"total"`
}

type ClassRecordExportCreateRequest struct {
	ExportType      string                `json:"exportType"`
	RecordIDs       []string              `json:"recordIds"`
	QueryConditions []ExportConditionItem `json:"queryConditions"`
}

type ClassRecordExportRecord struct {
	ID              int64                 `json:"id"`
	ExportType      string                `json:"exportType"`
	FileName        string                `json:"fileName"`
	ExporterName    string                `json:"exporterName"`
	TotalRows       int                   `json:"totalRows"`
	QueryConditions []ExportConditionItem `json:"queryConditions"`
	CreatedTime     *time.Time            `json:"createdTime,omitempty"`
	ExpiresAt       *time.Time            `json:"expiresAt,omitempty"`
	DownloadURL     string                `json:"downloadUrl,omitempty"`
}

File: government-admin/src/api/edu-center/class-record.ts
import axios from 'axios'
import { STORAGE_AUTHORIZE_KEY, useAuthorization } from '~/composables/authorization'
import { useGet, usePost } from '~/utils/request'

export interface ClassRecordQueryModel {
  beginStartTime?: string
  endStartTime?: string
  beginCreateTime?: string
  endCreateTime?: string
  beginUpdatedTime?: string
  endUpdatedTime?: string
  studentId?: string
  teacherIds?: string[]
  assistantTeacherIds?: string[]
  one2OneIds?: string[]
  timetableSourceTypes?: number[]
  studentSourceTypes?: number[]
  lessonChargingModeEnums?: number[]
  studentTeachingRecordStatuses?: number[]
  scheduleCallStatus?: number
  isArrear?: boolean | null
  lessonIds?: string[]
  classIds?: string[]
}

export interface StudentTeachingRecordItem {
  studentTeachingRecordId: string
  teachingRecordId: string
  studentId: string
  studentName: string
  studentPhone?: string
  avatar?: string
  teacherName?: string
  teacherEmployeeType?: number
  assistants?: string
  className?: string
  one2OneName?: string
  lessonName?: string
  status: number
  sourceType: number
  startTime?: string
  endTime?: string
  teachingRecordCreatedTime?: string
  timetableSourceType: number
  updatedTime?: string
  updatedStaffName?: string
  recordTime?: string
  quantity?: number
  actualQuantity?: number
  amount?: number
  skuMode?: number
  actualDeduct?: number
  actualTuition?: number
  arrearQuantity?: number
  remark?: string
  externalRemark?: string
  tuitionAccountId?: string
  tuitionAccountName?: string
  hasCompensated?: boolean
  subjectId?: string
  subjectName?: string
  advisorStaffId?: string
  advisorStaffName?: string
  studentManagerId?: string
  studentManagerName?: string
  teachingContent?: string
  teachingContentImages?: string[]
  classRoomName?: string
  one2OneTeachers?: string
  classTeachers?: string
  rollCallClassTeachers?: string
  currentClassTeachers?: string
}

export interface StudentTeachingRecordPagedResult {
  totalClassTimes?: number
  totalTuition?: number
  totalStudentCount?: number
  list?: StudentTeachingRecordItem[]
  total?: number
}

export interface TeachingRecordDetailTeacher {
  teacherId: string
  teacherName: string
  type: number
  status: number
  quantity: number
}

export interface TeachingRecordDetailStudent {
  studentTeachingRecordId: string
  studentId: string
  studentName: string
  studentPhone?: string
  avatar?: string
  sex?: number
  birthday?: string
  status: number
  sourceType: number
  quantity?: number
  actualQuantity?: number
  remark?: string
  externalRemark?: string
  isComment?: boolean
  isParentRead?: boolean | null
  tuitionAccountId?: string
  tuitionAccountName?: string
  isTuitionAccountActive?: boolean
  leftQuantity?: number
  skuMode?: number
  amount?: number
  actualDeduct?: number
  actualTuition?: number
  arrearQuantity?: number
  recordTime?: string
  updatedTime?: string
  updatedStaffName?: string
}

export interface TeachingRecordDetailResult {
  teachingRecordId: string
  sourceName?: string
  sourceType?: number
  sourceId?: string
  lessonId?: string
  lessonType?: number
  startTime?: string
  endTime?: string
  shouldAttendanceCount?: number
  actualAttendanceCount?: number
  leaveCount?: number
  truancyCount?: number
  teacherClassTime?: number
  studentTotalClassTime?: number
  studentActualTuition?: number
  teacherList?: TeachingRecordDetailTeacher[]
  studentList?: TeachingRecordDetailStudent[]
  createdTime?: string
  createdStaffName?: string
  timetableSourceType?: number
  classRoomName?: string
  classRoomId?: string
  defaultStudentClassTime?: number
  timetableSourceId?: string
  lessonName?: string
  teachingContent?: string
  subjectId?: string
  subjectName?: string
  teachingContentImages?: string[]
}

export interface UpdateStudentTeachingRecordParams {
  studentTeachingRecordId?: string
  teachingRecordId?: string
  studentId?: string
  sourceType?: number
  status: number
  quantity: number
  remark?: string
  externalRemark?: string
}

export interface UpdateTeachingRecordClassInfoParams {
  teachingRecordId: string
  teacherId: string
  assistantIds?: string[]
  classRoomId?: string
  teacherClassTime: number
}

export interface DeleteStudentTeachingRecordParams {
  studentTeachingRecordId: string
}

export interface ScheduleTeachingRecordItem {
  teachingRecordId: string
  timetableSourceId?: string
  startTime?: string
  endTime?: string
  timetableSourceType: number
  sourceName?: string
  sourceType?: number
  sourceId?: string
  lessonId?: string
  className?: string
  one2OneName?: string
  lessonName?: string
  subjectId?: string
  subjectName?: string
  rollCallStatus: number
  attendanceRate?: number
  attendCount?: number
  shouldAttendCount?: number
  leaveCount?: number
  absentCount?: number
  unrecordedCount?: number
  actualQuantity?: number
  actualTuition?: number
  teacherId?: string
  teacherName?: string
  assistants?: string
  classRoomName?: string
  teacherClassTime?: number
  commentCount?: number
  unCommentCount?: number
  readCount?: number
  unReadCount?: number
  createdTime?: string
  updatedTime?: string
}

export interface ScheduleTeachingRecordPagedResult {
  totalClassTimes?: number
  totalTeacherTimes?: number
  totalTuition?: number
  list?: ScheduleTeachingRecordItem[]
  total?: number
}

export interface ClassCommentQueryModel {
  teachingStartTime?: string
  teachingEndTime?: string
  teachingRecordTypes?: number[]
  lessonId?: string
  teacherIds?: string[]
  classId?: string
  one2OneId?: string
  classTeacherIds?: string[]
  one2OneTeacherIds?: string[]
}

export interface ClassCommentItem {
  teachingRecordId: string
  sourceName?: string
  sourceType?: number
  sourceId?: string
  lessonId?: string
  lessonName?: string
  createdTime?: string
  teacherId?: string
  teacherName?: string
  startTime?: string
  endTime?: string
  readCount?: number
  unReadCount?: number
  commentCount?: number
  unCommentCount?: number
  assistants?: string
  classRoomName?: string
}

export interface ClassCommentPagedParams {
  queryModel: ClassCommentQueryModel
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
  sortModel?: {
    startTime?: number
  }
}

export interface ClassCommentPagedResult {
  list?: ClassCommentItem[]
  total?: number
}

export interface ClassCommentStudentQueryModel {
  isParentFeedback?: boolean
  teachingStartTime?: string
  teachingEndTime?: string
  teachingRecordTypes?: number[]
  lessonId?: string
  teacherIds?: string[]
  assistantTeacherIds?: string[]
  classId?: string
  one2OneId?: string
  isComment?: boolean
  isRead?: boolean
  classTeacherIds?: string[]
  one2OneTeacherIds?: string[]
  studentId?: string
}

export interface ClassCommentStudentItem {
  teachingRecordId: string
  studentTeachingRecordId: string
  sourceName?: string
  sourceType?: number
  sourceId?: string
  lessonId?: string
  lessonName?: string
  teacherId?: string
  teacherName?: string
  startTime?: string
  endTime?: string
  avatar?: string
  studentName?: string
  studentId?: string
  studentPhone?: string
  isComment?: boolean
  isRead?: boolean | null
  assistants?: string
  classRoomName?: string
  isParentFeedback?: boolean
  parentFeedbackType?: number
  parentFeedbackGrade?: number
  parentFeedbackContent?: string
}

export interface ClassCommentStudentPagedParams {
  queryModel: ClassCommentStudentQueryModel
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
  sortModel?: {
    startTime?: number
  }
}

export interface ClassCommentStudentPagedResult {
  list?: ClassCommentStudentItem[]
  total?: number
}

export interface ClassRecordExportConditionItem {
  label: string
  value: string
}

export interface ClassRecordExportRecord {
  id: number
  exportType: string
  fileName: string
  exporterName: string
  totalRows: number
  queryConditions: ClassRecordExportConditionItem[]
  createdTime?: string
  expiresAt?: string
  downloadUrl?: string
}

export interface ClassRecordPagedParams {
  queryModel: ClassRecordQueryModel
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
  sortModel?: {
    startTime?: number
    updatedTime?: number
  }
}

export function getStudentTeachingRecordPagedListApi(data: ClassRecordPagedParams) {
  return usePost<StudentTeachingRecordPagedResult>('/api/v1/class-records/student-paged-list', data)
}

export function getScheduleTeachingRecordPagedListApi(data: ClassRecordPagedParams) {
  return usePost<ScheduleTeachingRecordPagedResult>('/api/v1/class-records/schedule-paged-list', data)
}

export function getClassCommentPagedListApi(data: ClassCommentPagedParams) {
  return usePost<ClassCommentPagedResult>('/api/v1/class-comments/paged-list', data)
}

export function getClassCommentStudentPagedListApi(data: ClassCommentStudentPagedParams) {
  return usePost<ClassCommentStudentPagedResult>('/api/v1/class-comments/student-paged-list', data)
}

export function getTeachingRecordDetailApi(params: { teachingRecordId: string }) {
  return useGet<TeachingRecordDetailResult>('/api/v1/class-records/detail', params)
}

export function updateTeachingRecordClassInfoApi(data: UpdateTeachingRecordClassInfoParams) {
  return usePost<boolean>('/api/v1/class-records/update-class-info', data)
}

export function exportClassRecordsApi(data: {
  exportType: string
  recordIds: string[]
  queryConditions: ClassRecordExportConditionItem[]
}) {
  return usePost<ClassRecordExportRecord>('/api/v1/class-records/export', data)
}

export function getClassRecordExportRecordsApi(exportType: string) {
  return useGet<ClassRecordExportRecord[]>('/api/v1/class-records/export-records', { exportType })
}

export async function downloadClassRecordExportRecordApi(recordId: number | string, exportType: string) {
  const token = useAuthorization()
  const response = await axios.get('/api/v1/class-records/export-records/download', {
    params: { recordId, exportType },
    responseType: 'blob',
    headers: {
      [STORAGE_AUTHORIZE_KEY]: token.value || '',
      Authorization: token.value ? `Bearer ${token.value}` : '',
      'Accept-Language': 'zh-CN',
    },
  })
  return response
}

export function updateStudentTeachingRecordApi(data: UpdateStudentTeachingRecordParams) {
  return usePost<boolean>('/api/v1/class-records/student/update', data)
}

export function deleteStudentTeachingRecordApi(data: DeleteStudentTeachingRecordParams) {
  return usePost<boolean>('/api/v1/class-records/student/delete', data)
}

export function deleteTeachingRecordApi(data: { teachingRecordId: string }) {
  return usePost<boolean>('/api/v1/class-records/delete', data)
}
